independent-projects/resteasy-reactive/common/processor/src/main/java/org/jboss/resteasy/reactive/common/processor/EndpointIndexer.java

```diff
@@ -822,6 +822,8 @@ public PARAM extractParameterInfo(ClassInfo currentClassInfo, ClassInfo actualEn
             builder.setType(ParameterType.HEADER);
             convertible = true;
         } else if (restHeaderParam != null) {
+            if (restHeaderParam.value() == null)
+                sourceName = StringUtil.hyphenate(sourceName);
             builder.setName(valueOrDefault(restHeaderParam.value(), sourceName));
```

Comment:
Don't overwrite sourceName, this is a pain to debug. Better do this instead:

```suggestion
            if (restHeaderParam.value() == null) {
            	builder.setName(StringUtil.hyphenate(sourceName));
            } else {
            	builder.setName(restHeaderParam.value().asString());
            }
```

Reply:
I see, as we won't be using `valueOrDefault` I'll add a check for empty values as well. 

```            
if (restHeaderParam.value() == null || restHeaderParam.value().asString().isEmpty()) {
                builder.setName(StringUtil.hyphenate(sourceName));
}
```